### PR TITLE
add commments in post

### PR DESCRIPTION
**Pull request by ShivanshSinghnnovative**

### CODE REVIEW

File: src/store/post-store.js [hunk 1]
```diff
@@ -42,10 +42,10 @@ export const postStore = (() => {
         }
 
     };
-
+    let posts = []
```
Comment (yuvrajInno) [inline src/store/post-store.js]: use state for these

File: src/store/post-store.js [hunk 2]
```diff
@@ -60,5 +60,66 @@ export const postStore = (() => {
             console.log(error)
         }
     }
-    return { addPost, getAllPosts };
+    const postComment = async (commentText, postId) => {
```
Comment (yuvrajInno) [inline src/store/post-store.js]: change function name

File: src/store/post-store.js [hunk 3]
```diff
@@ -60,5 +60,66 @@ export const postStore = (() => {
...
+        try {
+            const commentTitle = commentText;
+            console
```
Comment (yuvrajInno) [inline src/store/post-store.js]: remove these

File: src/store/post-store.js [hunk 4]
```diff
@@ -60,5 +60,66 @@ export const postStore = (() => {
...
+        }
+    };
+    const updateComment = async (createdAt, updatedComment, postId) => {
+        try {
+            const userDetailsCollection = collection(db, `post/${postId}/comment`);
+            const querySnapshot = await getDocs(query(userDetailsCollection, where("createdAt", "==", createdAt)));
+            if (!querySnapshot.empty) {
+                const userDocRef = querySnapshot.docs[0].ref;
+                await setDoc(userDocRef, updatedComment, { merge: true });
+                console.log("Comment updated successfully");
```
Comment (yuvrajInno) [inline src/store/post-store.js]: remove unnecessary consoles

File: src/pages/showPost.vue [hunk 5]
```diff
@@ -24,17 +24,56 @@
...
+                                    </v-icon>
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+                    <input v-model="comment[post.id]"
+                        class="border break-all border-gray-300 text-gray-800 text-sm rounded-lg mt-1  block w-4/5 p-2.5 bg-blue-200 "
+                        type="text" placeholder="add comments">
+                    <v-btn
+                        @click=" (comment[post.id] && editableCommentId[post.id] ? handelupdateComment(comment[post.id], post.id) : postCommentHandle(post.id))"
```
Comment (yuvrajInno) [inline src/pages/showPost.vue]: change function name

File: src/pages/showPost.vue [hunk 6]
```diff
@@ -52,39 +91,46 @@
...
+const comment = ref({})
+const { getAllPosts, postComment, getCommentsForPost, updateComment, deleteComment } = userPosts;
 const tagUsers = ref({});
 const posts = ref([]);
-const toggleTagUser = (postId) => {
-    tagUsers.value[postId] = !tagUsers.value[postId];
-};
-const visiblePostCount = ref(5);
-const totalPosts = ref(0);
+const comments = ref({});
```
Comment (yuvrajInno) [inline src/pages/showPost.vue]: change name to postCommentList

File: src/pages/showPost.vue [hunk 7]
```diff
@@ -93,4 +139,44 @@ const truncateDescription = (description, lines) => {
...
+}
+const editComment = (postId, commentId, createdAt) => {
+    editableCommentId.value[postId] = commentId;
+    console.log(editableCommentId.value, postId)
+    comment.value[postId] = allcomments.value[postId].find(comment => comment.id === commentId).commentTitle;
+    commentsToEdit.value[postId] = createdAt
+
+};
+
+const handelupdateComment = async (updateExistingComment, postId) => {
```
Comment (yuvrajInno) [inline src/pages/showPost.vue:169]: add trim function

File: src/store/post-store.js [hunk 8]
```diff
@@ -60,5 +60,61 @@ export const postStore = (() => {
             console.log(error)
         }
     }
-    return { addPost, getAllPosts };
+    const addCommentOnPost = async (commentText, postId) => {
+        try {
+            const commentTitle = commentText;
```
Comment (meanstack9716) [inline src/store/post-store.js]: do you really need this const commentTitle = commentText;?
you can directly change variable name in argument declaration @ShivanshSinghnnovative

**Review by meanstack9716 — CHANGES_REQUESTED** [review]

File: src/store/post-store.js [hunk 9]
```diff
@@ -60,5 +60,60 @@ export const postStore = (() => {
...
+            });
+            return comments;
+        } catch (error) {
+            console.error(error);
+        }
+    };
+    const updateComment = async (createdAt, updatedComment, postId) => {
+        try {
+            const userDetailsCollection = collection(db, `post/${postId}/comment`);
+            const querySnapshot = await getDocs(query(userDetailsCollection, where("createdAt", "==", createdAt)));
```
Comment (meanstack9716) [inline src/store/post-store.js]: why are you using createdAt  for where clause? do you not have the comment UUID? @ShivanshSinghnnovative

File: src/store/post-store.js [hunk 10]
```diff
@@ -60,5 +60,60 @@ export const postStore = (() => {
...
+                comments.push(comment);
+            });
+            return comments;
+        } catch (error) {
+            console.error(error);
+        }
+    };
+    const updateComment = async (createdAt, updatedComment, postId) => {
+        try {
+            const userDetailsCollection = collection(db, `post/${postId}/comment`);
```
Comment (meanstack9716) [inline src/store/post-store.js]: why comment collection names on userDetailsCollection? @ShivanshSinghnnovative

**Review by meanstack9716 — CHANGES_REQUESTED** [review]

**Pull request merged**